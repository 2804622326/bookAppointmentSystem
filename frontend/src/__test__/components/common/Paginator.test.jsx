import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Paginator from '../../../components/common/Paginator.jsx';

test('renders pagination items and handles click', () => {
  const onPaginate = jest.fn();
  const { asFragment } = render(
    <Paginator itemsPerPage={2} totalItems={5} currentPage={1} paginate={onPaginate} />
  );
  const page3 = screen.getByText('3');
  fireEvent.click(page3);
  expect(onPaginate).toHaveBeenCalledWith(3);
  expect(asFragment()).toMatchSnapshot();
});
