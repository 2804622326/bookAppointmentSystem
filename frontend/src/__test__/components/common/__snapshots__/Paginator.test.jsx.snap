// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders pagination items and handles click 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-end"
  >
    <ul
      class="pagination"
    >
      <li
        class="page-item active"
      >
        <span
          class="page-link"
        >
          1
          <span
            class="visually-hidden"
          >
            (current)
          </span>
        </span>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
          tabindex="0"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
          tabindex="0"
        >
          3
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
